src/devices/Dhtxx/DhtSensor.cs

```diff
@@ -0,0 +1,218 @@
+
+using System;
+using System.Device.Gpio;
+using System.Diagnostics;
+
+namespace Iot.Device.DHTxx
+{
+    public class DelayMS
+    {
+
+        private Stopwatch stopwatch = Stopwatch.StartNew();
+
+        /// <summary>
+        /// Wait asyncrhonous way for a specific number of milliseconds
+        /// 
+        /// </summary>
+        /// <param name="milliseconds">Number of milliseconds to wait</param>
+        /// <remarks>
+        /// This function doesn't work if you want to wait for less than 100 microseconds.
+        /// </remarks>
+        public void Wait(double milliseconds)
+        {
+            long initialTick = stopwatch.ElapsedTicks;
+            long initialElapsed = stopwatch.ElapsedMilliseconds;
+            double desiredTicks = milliseconds / 1000.0 * Stopwatch.Frequency;
+            double finalTick = initialTick + desiredTicks;
+            while (stopwatch.ElapsedTicks < finalTick)
+            {
+                //nothing than waiting
+            }
+        }
+
+    }
+
+    public class DHTSensor
+    {
+        private byte[] dht11_val = new byte[5];
+        private int pin;
+        private DhtType dhtType;
+        const int MAX_TIME = 85;
+        const uint MAX_WAIT = 255;
+        
+        /// <summary>
+        /// How last read went
+        /// </summary>
+        public bool LastRead { get; internal set; }
```

Comment:
I'd rename this to LastReadSuccessful or remove it and change other places to something like i.e.:
`bool TryGetTemperature()`